### PR TITLE
[VSTS] --no-sudo option in system-dependencies

### PR DESCRIPTION
**Pull request by joewstroman**
I removed the product signing scripts and only have changes specific to this repo (provisioning and packaging). Eventually I'll be adding Bockover's provisionator bootstrap scripts, but these preliminary changes will provide green master builds on VSTS.

https://devdiv.visualstudio.com/DevDiv/Default/_build/index?buildId=956611&_a=summary

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -175,6 +175,16 @@ else
...
+package:
+	mkdir -p ../package
+	$(MAKE) -C ../maccore package
+	# copy .pkg, .zip and *updateinfo to the packages directory to be uploaded to storage
+	cp ../maccore/release/*.pkg ../package
+	-cp ../maccore/release/*.zip ../package
+	-cp ../maccore/release/*updateinfo ../package
+	-cp ../maccore/tests/*.zip ../package
+	-cp ../xamarin-macios/tests/*.zip ../package
+
```
Comment (spouliot) [inline Makefile]: this seems unrelated to the `--no-sudo` option

Comment (joewstroman) [reply Makefile]: Right the pull request title wasn't all encompassing. This was a part of the first pull request I submitted but I opened a new one with fewer changes. I'm building on VSTS and it makes it simpler to have the macios packaging step inside the macios makefile. We could be deprecating release-scripts and I believe this is the only repo that uses this task.

Comment (spouliot) [reply Makefile]: @joewstroman ok, please add this information to the commit message when merging. thanks!
